### PR TITLE
Add `resume`

### PR DESCRIPTION
**Pull request by natefaubion**
I also inlined the implementation of `runFree` so there's no `Identity` overhead.

### CODE REVIEW

File: src/Control/Monad/Free.purs [hunk 1]
```diff
@@ -132,6 +137,16 @@ runFreeM k = tailRecM go
...
     Bind g i -> Left <$> k (i <$> g)
 
+-- | Unwraps a single layer of the functor `f`.
+resume
+  :: forall f a
+   . (Functor f)
+  => Free f a
+  -> Either (f (Free f a)) a
+resume f = case toView f of
+  Return a → Right a
```
Comment (paf31) [inline src/Control/Monad/Free.purs]: I'd like to avoid using unicode in core libraries.

File: src/Control/Monad/Free.purs [hunk 2]
```diff
@@ -132,6 +137,16 @@ runFreeM k = tailRecM go
     Return a -> Right <$> pure a
     Bind g i -> Left <$> k (i <$> g)
 
+-- | Unwraps a single layer of the functor `f`.
+resume
+  :: forall f a
+   . (Functor f)
```
Comment (paf31) [inline src/Control/Monad/Free.purs]: Parens are redundant here.

File: src/Control/Monad/Free.purs (hunk 1 above)
Comment (natefaubion) [inline src/Control/Monad/Free.purs]: Hah! Sorry about that. It's habit now...

**Comment by paf31** [comment]
👍 LGTM, thanks!

**Comment by natefaubion** [comment]
Github is not reflecting my changes here for some reason. I'm going to try to close and reopen.

**Pull request closed without merging**